src/domain/user/services/UserCreateService.ts

```diff
@@ -1,17 +1,38 @@
-import Mock from '../mocks/UserMock';
-import Iuser from '../types/UserTypes';
-import UserSchema from '../helper/JoiHelper';
+import { inject, injectable } from 'tsyringe'
+import { IUserCreate } from '../../../interface/domain/services/UserCreateTypes'
+import { IUtil } from '../../../interface/util/UtilTypes'
+import { IHelper } from '../../../interface/domain/helper/HelperTypes'
+import IUser from '../../../interface/UserTypes'
+import { IUserRepository } from '../../../interface/domain/Repository/RepositoryTypes'
 
-class UserService {
-  static async UserCreate(dados: Iuser) {
+@injectable()
+class UserService implements IUserCreate {
+  util : IUtil
+  userHelper : IHelper
+  userRepository : IUserRepository
+
+  constructor (
+    @inject('Util')util : IUtil,
+    @inject('UserHelper')userHelper : IHelper,
+    @inject('UserRepository')userRepository : IUserRepository) {
+    this.util = util
+    this.userHelper = userHelper
+    this.userRepository = userRepository
+  }
+
+  UserCreate (dados: IUser) {
     try {
-      const newUser = await UserSchema.validateAsync(dados);
-      Mock.push(newUser);
-      return { code: 201, msg: { message: 'usuario criado' } };
+      this.util.cpfCheck(dados.cpf)
+      this.userHelper.emailCheck(dados.email, this.userRepository.database)
```

Comment:
o ideal é retornar um booleano do helper, captar esse booleano e fazer um throw aqui na service caso esse boolean for false